security/backends/elasticsearch/throttling.py

```diff
@@ -0,0 +1,77 @@
+from datetime import timedelta
+
+from django.conf import settings
+from django.utils import timezone
+from django.utils.translation import ugettext as _
+from django.urls import resolve
+
+from elasticsearch_dsl import Q
+
+from ipware.ip import get_client_ip
+
+from security.enums import InputRequestSlug
+from security.throttling.validators import ThrottlingValidator
+from security.throttling.exception import ThrottlingException
+
+from .models import InputRequestLog
+
+
+class PerRequestThrottlingValidator(ThrottlingValidator):
```

Comment:
For these throttling classes, I think they should not be imported directly, but there should be some factory function/class, that will return the right validator according to used backend. From the prespective of project code, I think you don't need to worry which log backend is used in settings. Also, how do you determine, which validator to use when you have multiple backends is use? (for example elastic and SQL)

Reply:
Yes you are true. This is next step. I was thinging about the same thing. But again I cannot do whole change in one pull request. Therefore I have next tasks which will be solve it. But yes this is very good point